virtcontainers/utils/utils.go

```diff
@@ -259,3 +260,21 @@ func ValidCgroupPath(path string) string {
 	// clean up path and return a new path relative to defaultCgroupPath
 	return filepath.Join(DefaultCgroupPath, filepath.Clean("/"+path))
 }
+
+// IsProcRunning check if the process is running
+func IsProcRunning(pid int) (bool, error) {
+	if pid <= 0 {
+		return false, nil
+	}
+
+	process, err := os.FindProcess(pid)
+	if err != nil {
+		return false, err
+	}
+
+	if err := process.Signal(syscall.Signal(0)); err != nil {
```

Comment:
This looks racy to me - by the time `process.Signal()` has been called, the process might be dead.

Why not just call `unix.Kill(pid, 0)` and handle `ESRCH`?

Reply:
This code was origin from `isShimRunning`.  I think it can be replace by `unix.Kill(pid, 0)`